vendor_blueprints/export_card_to_file.py

```diff
@@ -0,0 +1,166 @@
+import json
+import sys
+import argparse
+import requests
+import urllib.parse
+import shipyard_utils as shipyard
+
+
+EXIT_CODE_INVALID_CREDENTIALS = 200
+EXIT_CODE_INVALID_ACCOUNT = 201
+EXIT_CODE_BAD_REQUEST = 202
+EXIT_CODE_INCORRECT_CARD_TYPE = 203
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--email', dest='client_id', required=True)
+    parser.add_argument('--password', dest='password', required=True)
+    parser.add_argument('--domo-instance', dest='domo_instance', required=True)
+    parser.add_argument('--card-id', dest='card_id', required=True)
+    parser.add_argument('--dest-file-name', dest='dest_file_name', required=True)
+    parser.add_argument('--dest-folder-path', 
+                        dest='dest_folder_path', 
+                        default='',
+                        required=True)
+    parser.add_argument('--file-type',
+                        dest='file_type',
+                        choices={'PPT', 'CSV', 'EXCEL'},
+                        required=True)
+    args = parser.parse_args()
+    return args
+
+
+def get_access_token(email, password, DOMO_INSTANCE):
+    """
+    Generate Access Token for use with internal content APIs.
+
+    email (str): email address of the user on domo.com
+    password (str): login password of the user domo.com
+    """
+    auth_api = f"https://{DOMO_INSTANCE}.domo.com/api/content/v2/authentication"
+
+    auth_body = json.dumps({
+        "method": "password",
+        "emailAddress": email,
+        "password": password
+    })
+
+    auth_headers = {'Content-Type' : 'application/json'}
+    try:
+        auth_response = requests.post(auth_api, data=auth_body, 
+                                      headers=auth_headers)
+    except Exception as e:
+        print(f"Request error: {e}")
+        sys.exit(EXIT_CODE_BAD_REQUEST)
+
+    auth_response_json = auth_response.json()
+    if auth_response_json["success"] is False: # Failed to login
+        print(f"Authentication failed due to reason: {auth_response_json['reason']}")
+        sys.exit(EXIT_CODE_INVALID_CREDENTIALS)
+        
+    # else if the authentication succeeded
+    domo_token = auth_response_json['sessionToken']
+    return domo_token
+
+
+def get_card_data(card_id, access_token, DOMO_INSTANCE):
+    """
+    Get metadata and property information of a single card
+
+    card_id (str): The unique ID of the card
+    access_token: client access token. Use the get_access_token() function to retrieve the token.
+
+    Returns:
+    card_response -> dict with the metadata and details of the card.
+    """
+    card_info_api = f"https://{DOMO_INSTANCE}.domo.com/api/content/v1/cards?urns={card_id}&parts=metadata,properties&includeFiltered=true"
+    card_headers = {
+            'Content-Type' : 'application/json',
+            'x-domo-authentication': access_token
+        }
+    card_response = requests.get(url=card_info_api, headers=card_headers)
```

Comment:
Let's use the requests library to build out the url parameters, instead of doing it ourselves. For instance, don't use `urns={card_id}&...`. Instead use the url / query parameter to requests and send the dict there and have requests encode them. 
Doing the encoding this way could lead to mishandling values that have reserved characters.